pkgs/os-specific/linux/dpdk/default.nix

```diff
@@ -1,54 +1,38 @@
-{ stdenv, lib, kernel, fetchurl, pkgconfig, numactl, shared ? false }:
+{ stdenv, lib
+, kernel
+, fetchurl
+, pkgconfig, meson, ninja, cmake
+, libbsd, numactl, libbpf, zlib, libelf, jansson, openssl, libpcap
+, doxygen, python3
+, shared ? false }:
 
 let
-
   kver = kernel.modDirVersion or null;
-
   mod = kernel != null;
 
 in stdenv.mkDerivation rec {
-  name = "dpdk-${version}" + lib.optionalString mod "-${kernel.version}";
-  version = "19.08";
+  pname = "dpdk";
+  version = "19.08" + lib.optionalString mod "-${kernel.version}";
 
   src = fetchurl {
     url = "https://fast.dpdk.org/rel/dpdk-${version}.tar.xz";
     sha256 = "0xgrkip2aji1c7jy5gk38zzwlp5ap1s6dmbcag5dnyy3bmwvmp9y";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ numactl ] ++ lib.optional mod kernel.moduleBuildDependencies;
-
-  RTE_KERNELDIR = if mod then "${kernel.dev}/lib/modules/${kver}/build" else "/var/empty";
-  RTE_TARGET = "x86_64-native-linuxapp-gcc";
-
-  # we need sse3 instructions to build
-  NIX_CFLAGS_COMPILE = [ "-msse3" ];
-  hardeningDisable = [ "pic" ];
+  nativeBuildInputs = [ pkgconfig meson ninja cmake doxygen python3.pkgs.sphinx python3 ];
+  buildInputs = [ numactl libbsd libbpf zlib libelf jansson openssl.dev libpcap ] ++ lib.optional mod kernel.moduleBuildDependencies;
 
   postPatch = ''
-    cat >>config/defconfig_$RTE_TARGET <<EOF
-# Build static or shared libraries.
-CONFIG_RTE_BUILD_SHARED_LIB=${if shared then "y" else "n"}
-EOF
-  '' + lib.optionalString (!mod) ''
-    cat >>config/defconfig_$RTE_TARGET <<EOF
-# Do not build kernel modules.
-CONFIG_RTE_EAL_IGB_UIO=n
-CONFIG_RTE_KNI_KMOD=n
-EOF
+    patchShebangs config/arm
   '';
 
-  configurePhase = ''
-    make T=${RTE_TARGET} config
-  '';
-
-  installTargets = [ "install-runtime" "install-sdk" "install-kmod" ]; # skip install-doc
-
-  installFlags = [
-    "prefix=$(out)"
-  ] ++ lib.optionals mod [
-    "kerneldir=$(kmod)/lib/modules/${kver}"
-  ];
+  mesonFlags = [
+    "-Denable_docs=true"
+    "-Denable_kmods=${if kernel != null then "true" else "false"}"
+  ]
+  ++ lib.optional (shared == false) [ "-Ddefault_library=static" ]
+  ++ lib.optional stdenv.isx86_64 "-Dmachine=nehalem"
```

Comment:
do you think it is possible to make this configurable for the package? i might want to have avx512 support and maybe override the package then... (so that we can get -march=native -mtune=native -maxv512)

Reply:
I we could make what's passed at `-Dmachine=` configurable, or users can always override `mesonFlags`. What did you mean?

I would advice against `native` though, for reproducibility.

Reply:
You already approved means you'll do a followup PR?

Reply:
erm. i have a cusom dpdk package with full optimizations ( -mavx512f etc ). so actually i don't care about this package that much anymore. dpdk and it's perfomance rely heavily on compiler optimizations and architecture specific settings. so unless, one day, they have a fat binary like in hyperscan, i have to use a local package then.

Reply:
Okay :-)